[SPARK-28801][DOC][FOLLOW-UP] Setup links and address other review comments

dilipbiswal opened the pull request:
### What changes were proposed in this pull request?

- Sets up links between related sections.
- Add "Related sections" for each section.
- Change to the left hand side menu to reflect the current status of the doc.
- Other minor cleanups.

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes. 

### How was this patch tested?
Tested using jykyll build --serve

dilipbiswal committed:
fix1

dilipbiswal committed:
[SPARK-28801][DOC][FOLLOW-UP] Setup links and address other review comments

SparkQA commented:
**[Test build #117453 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/117453/testReport)** for PR 27371 at commit [`540d2e9`](https://github.com/apache/spark/commit/540d2e9b9b6d7bf319797fb3e64e444c440cd0cc).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/22211/
Test PASSed.

SparkQA commented:
**[Test build #117453 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/117453/testReport)** for PR 27371 at commit [`540d2e9`](https://github.com/apache/spark/commit/540d2e9b9b6d7bf319797fb3e64e444c440cd0cc).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/117453/
Test PASSed.

dilipbiswal commented:
@srowen I have added the links. Also, from the left side panel, i removed the dead links for now. 
FYI - We are missing doc for a few clauses now. I am writing them at present. We will add the doc and associated links going forward.

srowen commented on docs/_data/menu-sql.yaml:
```diff
@@ -123,37 +123,27 @@
               url: sql-ref-syntax-qry-select.html 
               subitems:
-                - text: DISTINCT Clause
-                  url: sql-ref-syntax-qry-select-distinct.html
```
I see, so these don't exist anymore?

dilipbiswal replied:
@srowen Yeah Sean. So when we setup this initially, i had created placeholder links for the topics i thought we should cover. As you pointed out, it's not a good idea to have these dead links. So i have removed them. I am writing "FROM clause" and "WINDOW clause" now. With that, i think we will have most of the clauses covered. If we think we need to take on some specific topic an explain in more detail then, we can add these along with the links. what do you think ?

srowen commented:
OK to remove links to whatever does not exist right now. Code freeze is in a few days. Any further PRs can add the doc and add back the link.

maropu commented on docs/_data/menu-sql.yaml:
```diff
@@ -123,37 +123,27 @@
-                  url: sql-ref-syntax-qry-select-join.html
-                - text: ORDER BY Clause 
-                  url: sql-ref-syntax-qry-select-orderby.html
+                - text: WHERE Clause 
```
The order of these clauses depends on something? Can we reorder them along with the main menu? https://github.com/apache/spark/blob/master/docs/sql-ref-syntax-qry-select.md

dilipbiswal replied:
@maropu Did you want me to re-order menu in the left hand side panel as per the "related clauses" in https://github.com/apache/spark/blob/master/docs/sql-ref-syntax-qry-select.md ? i.e 
```
- [WHERE Clause](sql-ref-syntax-qry-select-where.html)
- [GROUP BY Clause](sql-ref-syntax-qry-select-groupby.html)
- [HAVING Clause](sql-ref-syntax-qry-select-having.html)
- [ORDER BY Clause](sql-ref-syntax-qry-select-orderby.html)
- [SORT BY Clause](sql-ref-syntax-qry-select-sortby.html)
- [CLUSTER BY Clause](sql-ref-syntax-qry-select-clusterby.html)
- [DISTRIBUTE BY Clause](sql-ref-syntax-qry-select-distribute-by.html)
- [LIMIT Clause](sql-ref-syntax-qry-select-limit.html)
```

maropu replied:
Yea, can we?

dilipbiswal replied:
@maropu Sure... will do.

dilipbiswal replied:
@maropu I have reconciled the link order. Also i have added a very brief description in the entry page which looks like this : 
<img width="972" alt="Screen Shot 2020-01-27 at 10 35 17 PM" src="https://user-images.githubusercontent.com/14225158/73240922-527a6500-4155-11ea-94e3-dc84bc7e642b.png">

Please see if this looks okay to you or if we want to re-word or add anything ?

cc @huaxingao @srowen

maropu commented on docs/sql-ref-syntax-qry-select-where.md:
```diff
@@ -39,10 +39,11 @@ WHERE boolean_expression
+    (100, 'John', 30),
+    (200, 'Mary', NULL),
+    (300, 'Mike', 80),
+    (400, 'Dan',  50);
```
Thanks for this followup!

maropu reviewed (approved):
Looks fine to me except for one comment.

huaxingao commented:
LGTM. 
@dilipbiswal 
I quickly checked all the docs. There are still more things we need to fix: 
```
sql-ref-syntax-ddl.md
sql-ref-syntax-dml.md
sql-ref-syntax-qry.md
sql-ref-syntax-aux.md
sql-ref-syntax-aux-describe.md
sql-ref-syntax-aux-show.md
sql-ref-syntax-aux-conf-mgmt.md
sql-ref-syntax-aux-resource-mgmt.md
```
Currently, all of these have the following content:

![image](https://user-images.githubusercontent.com/13592258/73235994-a0876c80-4145-11ea-9675-6be6dc50de90.png)

Do you have time to fix these? If not, I will try to find some time.

dilipbiswal committed:
Code review

dilipbiswal commented:
@huaxingao 
> I quickly checked all the docs. There are still more things we need to fix

Yeah, i did notice it :-). It would be nice if we have a brief descriptive hook rather than just listing
the "related sections" (i see that we have done that for a few commands) for each of these.

I have taken a stab at "Queries" here in the recent commit. I will take a look at others
after i finish a couple of clauses. If you can get to them  quicker, then please go ahead :-)

SparkQA commented:
**[Test build #117466 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/117466/testReport)** for PR 27371 at commit [`b4b6612`](https://github.com/apache/spark/commit/b4b6612c5aa720a121295f69605bd0873068af30).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/22224/
Test PASSed.

SparkQA commented:
**[Test build #117466 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/117466/testReport)** for PR 27371 at commit [`b4b6612`](https://github.com/apache/spark/commit/b4b6612c5aa720a121295f69605bd0873068af30).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/117466/
Test PASSed.

srowen commented:
Merged to master

dilipbiswal commented:
Thanks a lot @srowen @maropu @huaxingao
